Adding insecure option to disable client certificate verification

jairajmahadev opened the pull request:
Added ```--insecure``` option. When provided, disables SSL certificate verification in axios. Fixes #2001

jairajmahadev commented:
How do I fix this issue thrown by codefactor?

kelson42 commented:
@jairajmahadev Thank you for your PR, this error from CodeFactor is not your fault and is not easy to fix, please just ignore it. Put the PR in ready to review when you will be so far.

jairajmahadev commented:
@kelson42 PR is ready for review. Also, i've added a ``` console.log ``` to test and see the value of the ```insecure``` variable. I had added a ```logger.warn``` log earlier, but the warning was not shown on the console. Need some help with that.

kelson42 commented on src/Downloader.ts:
```diff
@@ -106,7 +107,9 @@ class Downloader {
   private articleUrlDirector: URLDirector
   private mainPageUrlDirector: URLDirector
 
```
Why an empty line here?

kelson42 commented on src/Downloader.ts:
```diff
@@ -130,7 +136,7 @@ class Downloader {
       // HTTP agent pools with 'keepAlive' to reuse TCP connections, so it's faster
       httpAgent: new http.Agent({ keepAlive: true }),
-      httpsAgent: new https.Agent({ keepAlive: true }),
+      httpsAgent: new https.Agent({ keepAlive: true, rejectUnauthorized: !this.insecure }), // rejectUnauthorized: false disables SSL verification
```
You mean TLS (not SSL)

jairajmahadev replied:
yes, it disables the SSL certificate verification step in the TLS handshake. So should I just change it to say ```disables TLS```?

kelson42 replied:
Yes

kelson42 commented on src/parameterList.ts:
```diff
@@ -40,6 +40,7 @@ export const parameterDescriptions = {
   optimisationCacheUrl: 'Object Storage URL (including credentials and bucket name) to cache optimised media files',
   forceRender:
     'Force the usage of a specific API end-point/render, automatically chosen otherwise. Accepted values: [ VisualEditor, WikimediaDesktop. WikimediaMobile ]. More details at https://github.com/openzim/mwoffliner/wiki/API-end-points',
+  insecure: 'Allow insecure server connections when using HTTPS',
```
Better: `Skip HTTPS server authenticity verification step`

jairajmahadev replied:
Okay

kelson42 commented on src/sanitize-argument.ts:
```diff
@@ -35,8 +35,12 @@ export async function sanitize_all(argv: any) {
+    insecure,
   } = argv
 
+  // set insecure to false by default
```
This is quite clear, you don't need to put a comment here

kelson42 replied:
There is nothing to sanitize here. Unfortunately there is not a real place defined looks like for such an initialisation, but I would say either in `src/sanitize-argument.ts` or in Downloader constructor (like `--webp`)

jairajmahadev replied:
directly in Downloader constructor works

kelson42 reviewed (changes requested):
Almost good, only a few changes.

kelson42 commented on src/Downloader.ts:
```diff
@@ -116,6 +118,9 @@ class Downloader {
     this.apiUrlDirector = new ApiURLDirector(MediaWiki.actionApiUrl.href)
+    this.insecure = insecure
+
+    console.log('Insecure: ', this.insecure) // printing for testing purpose
```
Please remove that line

jairajmahadev replied:
done

kelson42 reviewed (changes requested):
Last small thing and we can merge

kelson42 reviewed (approved):
LGTM
